### ISSUE TITLE
Event Handler: ALB returns 502 for CORS Preflight configuration in ALBResolver

### ISSUE THREAD

**Issue by francispeabody** [awslabs/aws-lambda-powertools-python#1248]
### What were you searching in the docs?

This is either a bug or a documentation issue. I am looking at [this section](https://awslabs.github.io/aws-lambda-powertools-python/latest/core/event_handler/api_gateway/#pre-flight) of the documentation. The way I read the documentation is that if cors_config is provided, powertools will automatically handle preflight 'OPTIONS' requests. In other words, you do not need to write and code to handle options requests as long as you provide a cors_config in the ALBResolver constructor. This does not appear to be the case. When fronting a lambda with an ALB, the preflight request fails, even if cors_config is provided, unless the options request is explicitly handled in the lambda code.

### Is this related to an existing documentation section?

https://awslabs.github.io/aws-lambda-powertools-python/latest/core/event_handler/api_gateway/#pre-flight

### How can we improve?

```markdown
Clarification of the documentation for preflight requests for ALBs specifically. Also a code example of how to handle preflight requests for a lambda fronted by and ALB would be nice. Thanks!
```

### Acknowledgment

- [X] I understand the final update might be different from my proposed suggestion, or refused.

**Comment by heitorlessa** [comment]
hey @francispeabody, thanks for helping us improve our docs! I've just created a test and we're returning the CORS headers as expected - could you provide a sample or something we could try to reproduce?

We can deploy an ALB + Function to try later once we have more details.

Thank you!

![image](https://user-images.githubusercontent.com/3340292/173859203-3f6527cd-fc05-4877-835e-b803b9d3f5f0.png)

**Comment by heitorlessa** [comment]
Reproduced! @francispeabody, it's our bug. Changing the labels to bug - repro here: https://github.com/heitorlessa/pt-1248

**Comment by heitorlessa** [comment]
@francispeabody here's a workaround for now while I find which header is causing ALB to panic: https://github.com/heitorlessa/pt-1248/commit/67cc63e2d86c6b11400338a885332f1cfe649af9

**Comment by heitorlessa** [comment]
Found the issue - ALB apparently faults if the `body` is `null`. I'm gonna raise a ticket to the ALB team but in the meantime, we'll make a bug fix release. I just wanna make sure API Gateway doesn't have side effects with a empty bull (not null) before making this change.

**Comment by heitorlessa** [comment]
Confirmed, it's just ALB. [CORS Specification](https://fetch.spec.whatwg.org/#http-responses) also doesn't dictate whether the response body should be present, so it's more of a serialization issue in ALB - we'll change on our side for now and inform the ALB team internally.

Thank you so much @francispeabody .

---

**ALB**

```bash
15:26:19 ➜ pt-1248 git:(master!) http --follow --verbose OPTIONS $ENDPOINT
OPTIONS /hello HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Content-Length: 0
Host: pt-12-myloa-nmcfguurzny6-1668993808.eu-west-1.elb.amazonaws.com
User-Agent: HTTPie/2.4.0

# Response

HTTP/1.1 204 No Content
Access-Control-Allow-Headers: Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key
Access-Control-Allow-Methods: GET,OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Max-Age: 1
Connection: keep-alive
Date: Thu, 16 Jun 2022 13:26:22 GMT
Server: awselb/2.0
```


**API Gateway**

```bash
15:26:23 ➜ pt-1248 git:(master!) ENDPOINT="https://420cn1i5v5.execute-api.eu-west-1.amazonaws.com/Prod/hello"

15:26:39 ➜ pt-1248 git:(master!) http --follow --verbose OPTIONS $ENDPOINT
OPTIONS /Prod/hello HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Content-Length: 0
Host: 420cn1i5v5.execute-api.eu-west-1.amazonaws.com
User-Agent: HTTPie/2.4.0


HTTP/1.1 200 OK
Access-Control-Allow-Headers: Content-Type, X-Forwarded-For
Access-Control-Allow-Methods: GET
Access-Control-Allow-Origin: *
Access-Control-Max-Age: 1
Connection: keep-alive
Content-Length: 3
Content-Type: application/json
Date: Thu, 16 Jun 2022 13:26:42 GMT
Via: 1.1 83bc0649a33d85c1cf516bf48779a390.cloudfront.net (CloudFront)
X-Amz-Cf-Id: UExltMRxrJ75RmUg227tCf94qgpMEIGpESVKOewjmsOjt0EaS7cXOQ==
X-Amz-Cf-Pop: AMS1-C1
X-Cache: Miss from cloudfront
x-amz-apigw-id: T0Re4HyqjoEFcDg=
x-amzn-RequestId: 5defa35b-1fe9-4a82-b808-e09a71c248c5

{}
```

**Comment by heitorlessa** [comment]
hey @francispeabody the fix is now out within 1.26.2 version in PyPi/SAR. I've started the release for Lambda Layer version 21 - might take a few hours to hit all commercial regions we support.
